src/screens/profile/profile.tsx

```diff
@@ -0,0 +1,175 @@
+        query: USER_PROFILE,
+        variables: { id: route.params?.userId },
+    })
+
+    const user = profileResponse.data?.userById
+
+    if (!user) {
+        return <ActivityIndicator
+            {...null}
+            style={globalStyles.flex1}
+            color={colors.primary} />
+    }
+
+    const renderFeaturedBlock = () => {
+        const featuredBlock: Block = user?.featured
+        if (featuredBlock) {
+            return (
+                <View style={styles.blockContainer}>
+                    <Text style={styles.featuredBlockTitle}>Featured Block</Text>
+                    <ComponentDelegate component={JSON.parse(user?.featured?.embedDisplay)} />
+                </View>
+            )
+        } else if (currentUser.user.id === user.id) {
+            return (
+                <>
+                    <Text style={styles.featuredBlockTitle}>Featured Block</Text>
+                    <Button
+                        mode="contained"
+                        uppercase={false}
+                        style={styles.addButton}
+                        onPress={() => console.log('Pressed')}>
+                        Add
+                    </Button>
+                </>
+            )
+        } else {
+            return <Text style={styles.featuredBlockTitle}>This user has no featured block</Text>
+        }
+    }
+
+    const renderStarredItem = () => {
+        if (!user?.featured) {
+            return null
+        }
+
+        const starCount = user?.featured.starCount ?? 0
+        return (
+            <Button
+                mode="text"
+                compact
+                color={colors.starring}
+                style={styles.editButton}
+                uppercase={false}
+                onPress={() => handleStarring()}>
+                <View style={styles.starContainer}>
+                    <MaterialCommunityIcons
+                        name={user?.featured.starred ? 'star' : 'star-outline'}
+                        color={colors.starring}
+                        size={25} />
+                    <Caption style={styles.caption}> {starCount} {starCount === 1 ? 'star ' : 'stars '}</Caption>
+                </View>
+            </Button>
+        )
+    }
+
+    const handleStarring = () => {
+        const request: StarredRequest = { blockId: user.featured.id, starred: !user.featured.starred }
+        setStarred(request)
+    }
+
+    return (
+        <View style={globalStyles.flex1}>
+            <ScrollView bounces={false}>
+                <Card>
+                    <Card.Content style={styles.profileHeader}>
+                        <Avatar.Image style={styles.profilePhoto} size={100} source={Images.avatar} />
+                        <View style={styles.userInfo}>
+                            {user && <Text style={styles.username}>@{user.username}</Text>}
+                            {user && <Caption style={styles.caption}>{user.displayName ?? user.username}</Caption>}
```

Comment:
Could you swap these, so that the display name is largest?

Reply:
Sure, updated.